force-app/main/default/classes/ServiceService.cls

```diff
@@ -34,13 +75,18 @@ public with sharing class ServiceService {
                 contactId
             )
         ) {
-            programIds.add(engagement.Program__c);
-            result.get('engagements')
-                .add(convertObjectToOption(engagement, engageProgram));
+            if (activeProgramEngagementStages.contains(engagement.Stage__c)) {
```

Comment:
Since customers can type in values, do we want to be case insensitive. I think historically we haven't been.